src/plugin.ts

```diff
@@ -513,6 +569,8 @@ function insertCompiledTemplate<EnvSpecificOptions>(
     configFile: false,
   }) as t.File;
 
+  scopeLocals.locals.length = 0;
```

Comment:
reset scope locals and set it to the ones discovered at the end. As ast plugins could have removed previously added locals.